flink/v2.1/flink/src/test/java/org/apache/iceberg/flink/sink/dynamic/TestEvolveSchemaVisitor.java

```diff
@@ -99,20 +104,76 @@ public void testMakeTopLevelPrimitivesOptional() {
     assertThat(existingSchema.columns().stream().allMatch(Types.NestedField::isRequired)).isTrue();
 
     UpdateSchema updateApi = loadUpdateApi(existingSchema);
-    EvolveSchemaVisitor.visit(updateApi, existingSchema, new Schema());
+    EvolveSchemaVisitor.visit(TABLE, updateApi, existingSchema, new Schema(), PRESERVE_COLUMNS);
     Schema newSchema = updateApi.apply();
     assertThat(newSchema.asStruct().fields()).hasSize(14);
     assertThat(newSchema.columns().stream().allMatch(Types.NestedField::isOptional)).isTrue();
   }
 
+  @Test
+  public void testDropUnusedColumns() {
```

Comment:
Could we have a test when the dropped column is inside a struct, array or map?
Just to ensure that we traverse the hierarchy correctly

Reply:
Added.